portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,28 +57,34 @@
 import com.google.appengine.api.datastore.Query.SortDirection;
 import com.google.gson.Gson;
 import java.io.IOException;
+import java.lang.Math;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import org.owasp.encoder.Encode;
 
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   private UserComment comment;
   private static final int DEFAULT_MAX_COMMENTS = 10;
+  private int currentPageId = 1;
+  private boolean isLatestInputDangerous = false;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Query comment history from Datastore.
     Query commentHistoryQuery = new Query("UserComment").addSort("timestamp", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery commentHistory = datastore.prepare(commentHistoryQuery);
-    int totalComments = commentHistory.countEntities(FetchOptions.Builder.withDefaults());
+    List<Entity> commentHistoryList = commentHistory.asList(FetchOptions.Builder.withDefaults());
+    int totalComments = commentHistoryList.size();
+    int totalPages;
```

Comment:
Can you move the totalPages declaration to line 99? In general it's best practice to define a variable right before it's used.